src/Helper.cpp

```diff
@@ -128,6 +129,20 @@ namespace geopm
         return pieces;
     }
 
+    std::string string_join(const std::vector<std::string> &list,
+                            const std::string &delim)
+    {
+        std::ostringstream result;
+        if (list.size() > 0) {
+            auto back_it = list.end() - 1;
+            for (auto str_it = list.begin(); str_it != back_it; ++str_it) {
+                result << *str_it << delim;
+            }
+            result << *back_it;
+        }
```

Comment:
Given this is a vector not a list, could be simpler to just do ` for i=0..list.size()` and check  i!=0 for delimiter

Reply:
Probably would compile to the same object code, but this avoids a branch in the loop.

Reply:
The bigger issue is this function has no unit test and no doxygen :(